### DISCUSSION TITLE
Add a timeout function wrapper

### DISCUSSION THREAD

**Discussion by empz** [rayepps/radash#436] [Ideas]
A `timeout` function that takes an async function and a timeout number and will throw an error if the passed function takes longer than the timeout. Maybe even use `tryit` under the hood to return error.

```ts

const [err, data] = await timeout(5000, async () => { 
  // long running function
});

// err is an `Error` if the function doesn't resolve in less than 5000ms (maybe a custom `TimeoutError`?)

if (err instanceof TimeoutError) {
  // handle timeout
}

```

Not sure if the same function could be used for regular (non-async) functions too.

**Comment by aleclarson** [comment]
Hello @empz! The Radashi fork has a `timeout` function in the works (https://github.com/radashi-org/radashi/pull/250). You can use it today by installing `radashi@beta`. :v:
